### DISCUSSION TITLE
Difficulty Solving FEA Problem with Higher Numerical Values

### DISCUSSION THREAD

**Discussion by TLWise** [idaholab/moose#22180] [Q&A Modules: General] [answered]
I am using MOOSE to solve a numerical problem that involves solving a steady state pressure driven flow problem. I have had success in getting a numerical solutions for pressures in the range of 1e-4 Pa.For My actual problem I require the solutions for much higher pressure on the order of 1e7 Pa. For the expected flow velocities the relative pressure drop for each case is expected to be the same. I have come to realized the issue that I am experiencing is related to the residual errors of each variable being calculated. At the1e-4 pressure my initial residuals are as follows: T_fluid:  13.5975, velocity: 1.60189e-08, p_fluid:  6.19058e-06. At the 1e-4 pressure my initial residuals are: T_fluid:  13.5975, velocity: 448.259,  p_fluid:  56047.1. Since the problem is being solved as a steady state problem I do not have the ability to gradually increase the pressure. How can I resolve this solution issue without having to use a transient method using lots of computational resources?

**Comment by GiudGiud** [comment, marked as answer]
Hello

If you dont want to use a relaxation to steady state transient, then your options are to:
- initialize close to the steady solution
- use more robust solver methods

Guillaume

**Reply by TLWise** [reply]
Would a potential resolution to the problem be a steady state solution at a lower pressure, then the use of that solution scaled for all previously solved variables (velocity and pressure) as the initial guess to the higher pressure solve?

**Reply by GiudGiud** [reply]
It could work.

**Reply by GiudGiud** [reply]
There is NO guarantee though. Another common technique is to ramp down viscosity through a pseudo transient

**Comment by friedmud** [comment]
Firstly, using a "Transient" solve where you slowly step to steady state can actually be faster...

But setting that aside you should try to use some of our scaling options to get the residual / jacobian entries around the same size for all physics.

The first option is "manual" scaling.  This will scale the residual for a variable using syntax like:

```
[Variables]
  [myvar]
    order = FIRST
    family = LAGRANGE
    scaling = 1e-7
  []
[]
```

See the documentation here: https://mooseframework.inl.gov/syntax/Variables/index.html

The other option is automatic scaling.  You turn it on by setting `automatic_scaling = true` in your Executioner block.  You can see all of the options for automatic scaling at the bottom of the page under "Solver Variable Scaling Parameters" here: https://mooseframework.inl.gov/source/executioners/Steady.html

**Reply by TLWise** [reply]
Thank you. I have had some success using automatic scaling. Manual scaling resulted either in no solution due to convergence prior to the start of the iteration process due to very small initial residual error, or iteration divergence due to subsequent iteration not being scaled. I also changed the units so that the pressure (Pa to GPa) variable does not dominate the others, the result of this method is similar to the manual scaling, where there is no iterations due to low initial residual error.

My automatic scaling use has been with the transient version of my problem. I will try it on the steady state to see the result.
